### PR TITLE
Testing mutation functions wrong argument scenarios

### PR DESCRIPTION
**Pull request by anderson-joyle**
Issue #856 
Initial draft push to double-check:

- Error and Blank as ObjNull? 
- If(false,Blank()) as _Min?
- Add more test cases.
- What makes a DType.IsTable equals true?

### CODE REVIEW

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Collect.txt [hunk 1]
```diff
@@ -54,4 +54,29 @@ Errors: Error 8-11: Name isn't valid. 'Foo' isn't recognized.|Error 0-15: The fu
...
+>> Collect(Blank(), Blank())
+Errors: Error 8-15: Invalid argument type (ObjNull). Expecting a Table value instead.|Error 0-25: The function 'Collect' has some invalid arguments.
+
+>> Collect("", r2)
+Errors: Error 8-10: Invalid argument type (Text). Expecting a Table value instead.|Error 0-15: The function 'Collect' has some invalid arguments.
+
+>> Collect("", "")
+Errors: Error 8-10: Invalid argument type (Text). Expecting a Table value instead.|Error 12-14: Invalid argument type (Text). Expecting a Record value instead.|Error 0-15: The function 'Collect' has some invalid arguments.
+
+// !JYL! (_Min)???
```
Comment (MikeStall) [inline src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Collect.txt:80]: This seems very odd! 
What does canvas say?

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

**Comment by BruceHaley** [comment]
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Connectors.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Core.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Interpreter.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.LanguageServerProtocol.dll**
:heavy_check_mark: No Binary Compatibility issues for **Microsoft.PowerFx.Transport.Attributes.dll**

File: src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Collect.txt (hunk 1 above)
Comment (anderson-joyle) [reply src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Collect.txt]: This has been resolved.

**Review by MikeStall — APPROVED** [review]
:shipit:

**Pull request merged**